lib/kubeclient/common.rb

```diff
@@ -11,24 +11,10 @@ module ClientMixin
     def initialize_client(
       uri,
       path,
-      version = nil,
-      ssl_options: {
-        client_cert: nil,
-        client_key: nil,
-        ca_file: nil,
-        cert_store: nil,
-        verify_ssl: OpenSSL::SSL::VERIFY_PEER
-      },
-      auth_options: {
-        username: nil,
-        password: nil,
-        bearer_token: nil,
-        bearer_token_file: nil
-      },
-      socket_options: {
-        socket_class: nil,
-        ssl_socket_class: nil
-      }
+      version,
+      ssl_options: nil,
+      auth_options: nil,
+      socket_options: nil
```

Comment:
@grosser I have the feeling that the defaults were repeated here as well because other libraries (e.g. `openshift_client`) are calling this and they expect the defaults to be there. It's probably safer to put them back in (now that's it's much simpler with the constants you defined).